src/l1_fetcher.rs

```diff
@@ -23,9 +23,9 @@ use crate::{
 const MAX_RETRIES: u8 = 5;
 
 #[derive(Error, Debug)]
-pub struct L1GetLogsError;
+pub struct L1FetchError;
 
-impl fmt::Display for L1GetLogsError {
+impl fmt::Display for L1FetchError {
     fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
         write!(f, "failed to get logs")
     }
```

Comment:
Once above is implemented, it's simple to have separate error message for each enum-variant.